feat: add GHCR support for container registry

gabor-boros opened the pull request:
This PR adds GHCR support for the container registry, so GitHub can be used over DockerHub or AWS ECR. Also, the PR sets the generated images as job outputs, so the images can be referenced in later steps if needed.

The later one is particularly useful when someone wants to build a pipeline where service build jobs are running parallel and the `config.yml` file should be updated at once.

gabor-boros committed:
feat: add GHCR support for container registry

Signed-off-by: Gabor Boros <gabor@opencraft.com>

magajh commented on .github/workflows/build.yml:
```diff
@@ -245,6 +263,25 @@ jobs:
+          # Determine the target key for the service
+          if [ -z "$TARGET_KEY" ]; then
+            # If TARGET_KEY is not set (static image tag scenario), determine it from service
+            TARGET_KEY=$(python picasso/.github/workflows/scripts/get_service_target_key.py --service ${{ inputs.SERVICE }})
```
@gabor-boros I tested this in the static-tag scenario (`USE_DYNAMIC_IMAGE_TAG = false`) and the `Set job outputs for image tag and name` step is failing with:

```text
Run # Determine the target key for the service
python: can't open file '/home/runner/work/ednx-strains/ednx-strains/strains/teak/base/picasso/.github/workflows/scripts/get_service_target_key.py': [Errno 2] No such file or directory
```

This seems to be related to the job defaults
```yaml
defaults:
  run:
    working-directory: strains/${{ inputs.STRAIN_PATH }}
```

This command `python picasso/.github/workflows/scripts/get_service_target_key.py --service ${{ inputs.SERVICE }}` ends up resolving to `strains/<path>/picasso/...`, but the `picasso` repo is actually checked out at `${{ github.workspace }}/picasso`

I think to fix this we could either set `working-directory: ${{ github.workspace }}` for this step, or use an absolute path to the script

magajh reviewed (commented):
Thank you for this contribution! 

Regarding the “Show image name for service” part, we’re planning to apply a fix in the upcoming sprint so that the image tag push doesn’t fail (in the dynamic tag case) when multiple builds are running or when there’s a push to the repository in the middle of a build. That said, the change you’re proposing here still feels very useful as a safeguard for edge cases where updating the tag in the repo fails for some reason and the user still needs to see which tag was generated

gabor-boros committed:
refactor: remove image tag output

Signed-off-by: Gabor Boros <gabor@opencraft.com>

gabor-boros committed:
docs: add registry documentation

Signed-off-by: Gabor Boros <gabor@opencraft.com>

gabor-boros commented:
@magajh sorry, this was a "bit" delayed. I removed the image tag outputs as it is not used anymore and it makes no sense to contribute something that used by noone. Instead, I focused on the documentation and added a new section to the readme listing and describing the current registries.

gabor-boros commented:
@magajh / @felipemontoya to close the loop here, may I have a +1 on this? 😊

felipemontoya reviewed (commented):
To me this makes sense. I'll leave merging to the team

magajh reviewed (approved):
Thanks for this! Everything looks good
